src/adapters/adapter.js

```diff
@@ -15,8 +13,16 @@ export default class Adapter {
 
   constructor (options={}) {
     this.options = options;
-    this.id = options.id || uuid();
     this.status = Adapter.STATUS.UNINITIALIZED;
+    if (options.roleMapping) {
+      try {
+        this.roleMapping = JSON.parse(options.roleMapping);
```

Comment:
Switch to loading an object from `options`, rather than `JSON.parse` - we'll build JSON parsing into the environment-config loader changes.